lmp/build.sh

```diff
@@ -62,6 +62,11 @@ touch ${archive}/bitbake_debug.log ${archive}/bitbake_warning.log ${archive}/bit
 touch ${archive}/bitbake_global_env.txt ${archive}/bitbake_image_env.txt && chown builder ${archive}/bitbake_*_env.txt
 touch ${archive}/app-preload.log && chown builder ${archive}/app-preload.log
 touch ${archive}/tuf-root-fetch.log && chown builder ${archive}/tuf-root-fetch.log
+if [ ! -z "$OS_SELFTEST" ]; then
+	mkdir ${archive}/selftest && chown builder ${archive}/selftest
+	su builder -c $HERE/bb-selftest.sh
+	exit
```

Comment:
Exit will abort the script and the resulted data won't be uploaded, is that what you want here?

Reply:
No, I like to upload the artifacts but also return the exit code of the last command.
Perhaps I need to save the return code of the `$HERE/bb-selftest.sh` and run the `lmp/build.sh` until the end.

Reply:
I think the reproducible test is the only one that produce artifacts and all the others just write log on the terminal.
but I need to confirm this by running some other tests.

Reply:
@ricardosalveti I have confirmed in my `reproducible` factory and there is a `selftest` folder on the build artifacts. even when the build fails.
I think it works because the folder is already on `${archive}/selftest` so don't need to be move like some others.